src/cgt/src/ClockGating.cpp

```diff
@@ -0,0 +1,1479 @@
+             CGT,
+             "clock_gating",
+             1,
+             "Starting clock gating with parameters:");
+  debugPrint(logger_,
+             CGT,
+             "clock_gating",
+             1,
+             "Minimum instances per clock gate: {}",
+             min_instances_);
+  debugPrint(logger_,
+             CGT,
+             "clock_gating",
+             1,
+             "Maximum gate condition cover size: {}",
+             max_cover_);
+  debugPrint(logger_,
+             CGT,
+             "clock_gating",
+             1,
+             "Group registers: {}",
+             group_instances_ == GroupInstances::No ? "No"
+             : group_instances_ == GroupInstances::NetPrefix
+                 ? "Net prefix"
+                 : "Instance prefix");
+  rand_bits_.seed(0);  // For reproducibility
+
+  auto network = sta_->getDbNetwork();
+  network_builder_.init(logger_, network);
+
+  sta_->ensureGraph();
+  sta_->ensureLevelized();
+  sta_->searchPreamble();
+
+  // Initialize ABC
+  abc::Abc_Start();
+  abc_factory_->AddDbSta(sta_);
+  abc_library_ = std::make_unique<lext::AbcLibrary>(abc_factory_->Build());
+
+  abc::Abc_SclInstallGenlib(abc_library_->abc_library(),
+                            /*Slew=*/0,
+                            /*Gain=*/0,
+                            /*fUseAll=*/0,
+                            /*nGatesMin=*/0);
+  abc::Mio_LibraryTransferCellIds();
+
+  dump("pre");
+
+  using ClockNet = sta::Net;
+  std::vector<std::pair<ClockNet*, sta::Instance*>> found_insts;
+  found_insts.reserve(instances_.size());
+  for (const auto& inst_name : instances_) {
+    auto inst = network->findInstance(inst_name.c_str());
+    if (inst) {
+      auto clk_pin = getClockPin(network, inst);
+      if (!clk_pin) {
+        logger_->error(CGT,
+                       1,
+                       "Clock pin not found for instance '{}'",
+                       network->name(inst));
+        continue;
+      }
+      found_insts.emplace_back(network->net(clk_pin), inst);
+    } else {
+      logger_->warn(CGT, 2, "Instance '{}' not found in the design", inst_name);
+    }
+  }
+  if (found_insts.empty()) {
+    auto top = network->topInstance();
+    auto inst_iter = network->childIterator(top);
+    while (inst_iter->hasNext()) {
+      auto inst = inst_iter->next();
+      auto clk_pin = getClockPin(network, inst);
+      if (clk_pin) {
+        found_insts.emplace_back(network->net(clk_pin), inst);
+      }
+    }
+  }
+
+  std::unordered_map<std::string,
+                     std::vector<std::pair<ClockNet*, sta::Instance*>>>
+      inst_groups;
+  for (auto [clk_net, inst] : found_insts) {
+    std::string prefix = network->name(inst);
+    if (group_instances_ == GroupInstances::NetPrefix) {
+      auto out_pin = getRegOutPin(network, inst);
+      prefix = network->name(network->net(out_pin));
+    }
+    if (group_instances_ != GroupInstances::No) {
+      auto last_dot_pos = prefix.find_last_of('.');
```

Comment:
What is special about the '.'?  The option description `instance name prefix` doesn't make this clear.

Reply:
That doesn't mean it will be the hierarchy separator in odb.  I'm not really clear what you are trying to do here with the names.